### PR TITLE
Update Get started with Swashbuckle to 3.0

### LINKED ISSUE #13872: 3.0 update: getting-started-with-swashbuckle
getting-started-with-swashbuckle

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 14fce65f-1337-58cb-2193-8d8e0e62792d
* Version Independent ID: 59222460-a971-e76e-4fe9-23d4d7d768ee
* Content: [Get started with Swashbuckle and ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-swashbuckle?view=aspnetcore-3.0&tabs=visual-studio)
* Content Source: [aspnetcore/tutorials/getting-started-with-swashbuckle.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/tutorials/getting-started-with-swashbuckle.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @zuckerthoben
* Microsoft Alias: **scaddie**

### PR DESCRIPTION
**Pull request by spboyer**
- Fixes #13872 
- adds Swashbuckle/OpenAPI sample for docs.microsoft.com/samples 
- updates to 3.0

### CODE REVIEW

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md [hunk 1]
```diff
@@ -111,7 +127,11 @@ If using directories with IIS or a reverse proxy, set the Swagger endpoint to a
 Swagger provides options for documenting the object model and customizing the UI to match your theme.
 
 In the Startup class, add the following namespaces:
```
Comment (scottaddie) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md]:
```suggestion
In the `Startup` class, add the following namespaces:

```

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md [hunk 2]
```diff
@@ -83,16 +83,32 @@ Add the Swagger generator to the services collection in the `Startup.ConfigureSe
 
 ::: moniker-end
 
-::: moniker range=">= aspnetcore-2.1"
+::: moniker range="= aspnetcore-2.1"
```
Comment (scottaddie) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md]: I assume you want this to display for both 2.1 and 2.2. Consider the following suggestion all throughout the doc:

```suggestion
::: moniker range=">= aspnetcore-2.1 <= aspnetcore-2.2"
```

File: aspnetcore/tutorials/web-api-help-pages-using-swagger/samples/3.0/TodoApi.Swashbuckle/Program.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+
+namespace TodoApi
+{
+    public class Program
+    {
+        public static void Main(string[] args)
```
Comment (scottaddie) [inline aspnetcore/tutorials/web-api-help-pages-using-swagger/samples/3.0/TodoApi.Swashbuckle/Program.cs]: Let's use the newer pattern the 3.0 template uses for this class:

```csharp
        public static void Main(string[] args)
        {
            CreateHostBuilder(args).Build().Run();
        }

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    webBuilder.UseStartup<Startup>();
                });
```

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md (hunk 2 above)
Comment (spboyer) [reply aspnetcore/tutorials/getting-started-with-swashbuckle.md]: Made the change where possible. Wasn't aware of this syntax. Thanks!

File: aspnetcore/tutorials/web-api-help-pages-using-swagger/samples/3.0/TodoApi.Swashbuckle/Program.cs (hunk 3 above)
Comment (spboyer) [reply aspnetcore/tutorials/web-api-help-pages-using-swagger/samples/3.0/TodoApi.Swashbuckle/Program.cs]: Missed this file on the update. Thanks!

**Comment by spboyer** [comment]
@scottaddie - updated. Thanks for the quick review.

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md [hunk 4]
```diff
@@ -110,8 +126,12 @@ If using directories with IIS or a reverse proxy, set the Swagger endpoint to a
 
 Swagger provides options for documenting the object model and customizing the UI to match your theme.
 
-In the Startup class, add the following namespaces:
-[!code-csharp[](~/tutorials/web-api-help-pages-using-swagger/samples/2.0/TodoApi.Swashbuckle/Startup2.cs?name=snippet_PreReqNamespaces)]
+In the `Startup` class, add the following namespaces:
```
Comment (scottaddie) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md]:
```suggestion
In the `Startup` class, add the following namespaces:

```

File: aspnetcore/tutorials/getting-started-with-swashbuckle.md [hunk 5]
```diff
@@ -83,16 +83,32 @@ Add the Swagger generator to the services collection in the `Startup.ConfigureSe
...
 
+::: moniker range=">= aspnetcore-3.0"
+
+[!code-csharp[](../tutorials/web-api-help-pages-using-swagger/samples/3.0/TodoApi.Swashbuckle/Startup2.cs?name=snippet_ConfigureServices&highlight=8-11)]
+
+::: moniker-end
+
 In the `Startup.Configure` method, enable the middleware for serving the generated JSON document and the Swagger UI:
 
+::: moniker range="<= aspnetcore-2.1"
```
Comment (scottaddie) [inline aspnetcore/tutorials/getting-started-with-swashbuckle.md]:
```suggestion
::: moniker range=">= aspnetcore-2.1 <= aspnetcore-2.2"
```

File: aspnetcore/tutorials/web-api-help-pages-using-swagger/samples/3.0/TodoApi.Swashbuckle/Program.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.AspNetCore;
```
Comment (scottaddie) [inline aspnetcore/tutorials/web-api-help-pages-using-swagger/samples/3.0/TodoApi.Swashbuckle/Program.cs]: Replace these 3 `using` statements with the following:

```csharp
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Hosting;
```

**Comment by spboyer** [comment]
Ok, good to go now @scottaddie

**Comment by scottaddie** [comment]
Thank you, @spboyer!

**Pull request merged**